src/wfgoto.cpp

```diff
@@ -382,11 +383,64 @@ BOOL BuildDirectoryBagOValues(BagOValues<PDNODE> *pbov, vector<PDNODE> *pNodes,
 		AddBackslash(szPath);
 		lstrcat(szPath, lfndta.fd.cFileName);         // cFileName is ANSI now
 
-		// add directories in subdir
-		if (!BuildDirectoryBagOValues(pbov, pNodes, szPath, pNodeChild, scanEpoc))
+		// do not follow inner reparse points
+		BOOL bFollow = FALSE;
+		if (lfndta.fd.dwFileAttributes & FILE_ATTRIBUTE_REPARSE_POINT)
 		{
-			WFFindClose(&lfndta);
-			return FALSE;
+			// Check which reparsepoint
+			wchar_t szTemp[MAXPATHLEN];
+			DWORD tag = DecodeReparsePoint(szPath, szTemp, MAXPATHLEN);
+			switch (tag) 
+			{
+			// We always want to follow OneDrive
+			case IO_REPARSE_TAG_CLOUD:
+			case IO_REPARSE_TAG_CLOUD_1:
+			case IO_REPARSE_TAG_CLOUD_2:
+			case IO_REPARSE_TAG_CLOUD_3:
+			case IO_REPARSE_TAG_CLOUD_4:
+			case IO_REPARSE_TAG_CLOUD_5:
+			case IO_REPARSE_TAG_CLOUD_6:
+			case IO_REPARSE_TAG_CLOUD_7:
+			case IO_REPARSE_TAG_CLOUD_8:
+			case IO_REPARSE_TAG_CLOUD_9:
+			case IO_REPARSE_TAG_CLOUD_A:
+			case IO_REPARSE_TAG_CLOUD_B:
+			case IO_REPARSE_TAG_CLOUD_C:
+			case IO_REPARSE_TAG_CLOUD_D:
+			case IO_REPARSE_TAG_CLOUD_E:
+			case IO_REPARSE_TAG_CLOUD_F:
+				bFollow = TRUE;
+				break;
+
+			case IO_REPARSE_TAG_SYMLINK:
+			case IO_REPARSE_TAG_MOUNT_POINT:
+				// Check if it is an inner reparsepoint, because we don't want to follow 
+				// inner peparse points, because they would cause double enumeration
+
+				// If the anchor is not part of the path then it is an outer reparse point
+				// e.g. outer reparsepoint: szCachedRootLower == c:\bla, szTemp == d:\foo
+				// e.g. inner reparsepoint: szCachedRootLower == c:\bla, szTemp == c:\bla\foo
+				_wcslwr_s(szTemp, MAXPATHLEN);
+				if (!wcsstr(szTemp, szCachedRootLower))
+					bFollow = TRUE;
+				break;
+
+			default:
+				bFollow = FALSE;

```

Comment:
Since you are very detailed oriented, there should be a break; statement after setting bFollow to FALSE.  Fall through cases, if used, should be explicitly noted in a comment.

Reply:
done